lib/elasticity/base_document.rb

```diff
@@ -49,5 +42,41 @@ def delete
     def created?
       @created || false
     end
+
+    def self.elasticity_attributes(*args)
```

Comment:
A document doesn't only have top level attributes, it can contain nested attributes which was the main reason we never implemented something like this. I think this is the wrong abstraction for the problem and since this is a open/public project, it will be a lot harder to move away from this once introduced. 

The original plan was to dynamically generate the attribute accessors and the `to_document` method from the provided mapping. The mapping is the full definition of the document, including data types, and would be the right way to go but it's definitely a larger project/implementation.

Also, `from_active_record` is not a method that exists inside elasticity. Elasticity documents does not have dependency on ActiveRecord and it should be kept that way. At doximity `from_active_record` was introduced as a convention to allow a document to be created from an activerecord base but that's not part of the scope of the gem.